test/feature-benchmark/scenarios_skew.py

```diff
@@ -0,0 +1,71 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from math import floor
+from textwrap import dedent
+from typing import List
+
+from materialize.feature_benchmark.action import Action, TdAction
+from materialize.feature_benchmark.measurement_source import MeasurementSource, Td
+from materialize.feature_benchmark.scenario import Scenario
+
+
+def create_skewed_table(scale: int) -> str:
+    count = 10**scale
+    return (
+        dedent(
+            f"""
+       > CREATE TABLE skewed_table(f1 INTEGER, f2 INTEGER DEFAULT 1);
+       # Make sure 0 is overepresented
+       > INSERT INTO skewed_table (f1) SELECT 0 FROM generate_series(1, {count});
+       """
+        )
+        + "\n".join(
+            [
+                f"> INSERT INTO skewed_table (f1) SELECT MOD(generate_series, POW(10, {i})) FROM generate_series(1, {count} / {scale});"
```

Comment:
There are two ways that I used to increase the signal: (a) Use a join closure that does not consolidate well on the "right" side of the join, but does so on the left side; (b) Use a number of updates to the skewed keys to amplify the unconsolidated data that the join needs to process.

With (a), we force one implementation to emit more than the other, while with (b) we just have a convenient way to make the join have to process more rows. Also (b) can blow up the output even if the join is cardinality bounding.

Reply:
> (a) Use a join closure that does not consolidate well on the "right" side of the join, but does so on the left side; 

More specifics here: In the scenario I experimented with, the updates were on the uniform table and the materialized view returned a key for the skewed table. Without updates, the join would have been cardinality bounding, i.e., just a lookup through a foreign key column that does not change the cardinality of the uniform table. However, with updates, this does not hold anymore. Since the looked up and returned value was always the same, it consolidated really well with one implementation and not the other.